flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/LongHashJoinGenerator.scala

```diff
@@ -49,10 +49,13 @@ object LongHashJoinGenerator {
         keyType.getFieldCount == 1 && {
       keyType.getTypeAt(0).getTypeRoot match {
         case BIGINT | INTEGER | SMALLINT | TINYINT | FLOAT | DOUBLE | DATE |
-             TIME_WITHOUT_TIME_ZONE | TIMESTAMP_WITH_LOCAL_TIME_ZONE => true
+             TIME_WITHOUT_TIME_ZONE => true
         case TIMESTAMP_WITHOUT_TIME_ZONE =>
           val timestampType = keyType.getTypeAt(0).asInstanceOf[TimestampType]
           if (SqlTimestamp.isCompact(timestampType.getPrecision)) true else false
+        case TIMESTAMP_WITH_LOCAL_TIME_ZONE =>
```

Comment:
Bug here? You have not modified `genGetLongKey`?
Created a JIRA: https://issues.apache.org/jira/browse/FLINK-15049